sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -733,6 +724,33 @@ struct _pi_context : _pi_object {
   std::unordered_map<void *, MemAllocRecord> MemAllocs;
 
 private:
+  // Calculate root device.
+  pi_device getRootDevice() const {
+    if (Devices.size() == 1)
+      return Devices[0];
```

Comment:
can't this be a sub-device, not root-device?

Reply:
Explained in the comment of the method, it can be a sub-device. Please notice that this code was just moved from constructor, I didn't change it, so it is not related to this PR.